environments/contrail/install_vrouter_kmod.yaml

```diff
@@ -303,8 +304,13 @@ resources:
                   ip route add default via $def_gw
                 fi
               fi
-              if [[ ${dpdk_driver} == "vfio-pci" || ${sriov_on} == 'true' ]]; then
-                sed 's/^\(GRUB_CMDLINE_LINUX=".*\)"/\1 $KERNEL_ARGS intel_iommu=on iommu=pt"/g' -i /etc/default/grub ;
+              if [[ ${dpdk_driver} == "vfio-pci" || ${sriov_on} == 'True' ]]; then
+                if [[ ${dpdk_driver} == "vfio-pci" ]]; then
+                  $iommu_args='intel_iommu=on iommu=pt'
+                else
+                  $iommu_args='intel_iommu=on'
+                fi
+                sed 's/^\(GRUB_CMDLINE_LINUX=".*\)"/\1 $KERNEL_ARGS ${iommu_args}"/g' -i /etc/default/grub ;
```

Comment:
It is single quoted string, so no variables substitution,  are you sure it works?